nesta_ds_utils/file_ops.py

```diff
@@ -1,53 +1,151 @@
-from io import BytesIO
-from typing import Union
+import io
+from typing import Union, List
 from pathlib import Path
 from xmlrpc.client import Boolean
 import zipfile
 import os
+import boto3
+from fnmatch import fnmatch
+import pandas as pd
 
 
 def _convert_str_to_pathlib_path(path: Union[Path, str]) -> Path:
     """Converts a path written as a string to pathlib format.
 
-    :param path: file path in string format
-    :type path: str or pathlib.path
-    :return: path in pathlib format
-    :rtype: pathlib.Path
+    Args:
+        path (Union[Path, str]): File path in string format.
+
+    Returns:
+        Path: Path in pathlib format.
     """
     return Path(path) if type(path) is str else path
 
 
 def make_path_if_not_exist(path: Union[Path, str]):
     """Check if path exists, if it does not exist then create it.
 
-    :param path: file path 
-    :type path: pathlib.Path or str
+    Args:
+        path (Union[Path, str]): File path.
     """
     path = _convert_str_to_pathlib_path(path)
     if not path.exists():
         path.mkdir(parents=True)
 
 
 def extractall(
-    zip_path: Union[Path, str], 
-    out_path: Union[Path, str]=None, 
-    delete_zip: Boolean = True):
+    zip_path: Union[Path, str],
+    out_path: Union[Path, str] = None,
+    delete_zip: Boolean = True,
```

Comment:
(mis?)use of `Boolean`

Reply:
Yes, in line with comment above. I will take care of changing this too.